addons/website/views/website_templates.xml

```diff
@@ -1498,6 +1496,93 @@
     </xpath>
 </template>
 
+<!-- Additional colors option -->
+<!-- Primary -->
+<template id="template_header_mobile_additional_color_primary" inherit_id="website.template_header_mobile" active="False">
+    <xpath expr="//div[hasclass('o_main_nav')]" position="before">
+        <t t-set="_additional_btn_color" t-valuef="btn btn-outline-primary"/>
+    </xpath>
+</template>
+
+<template id="template_header_default_additional_color_primary" inherit_id="website.template_header_default" active="False">
+    <xpath expr="//div[hasclass('o_main_nav')]" position="before">
+        <t t-set="_additional_btn_color" t-valuef="btn btn-outline-primary"/>
+    </xpath>
+</template>
+
+<template id="template_header_hamburger_additional_color_primary" inherit_id="website.template_header_hamburger" active="False">
+    <xpath expr="//div[hasclass('o_main_nav')]" position="before">
+        <t t-set="_additional_btn_color" t-valuef="btn btn-outline-primary"/>
+    </xpath>
+</template>
+
+<template id="template_header_sales_two_additional_color_primary" inherit_id="website.template_header_sales_two" active="False">
+    <xpath expr="//div[hasclass('o_main_nav')]" position="before">
+        <t t-set="_additional_btn_color" t-valuef="btn btn-outline-primary"/>
+    </xpath>
+</template>
+
+<template id="template_header_sales_four_additional_color_primary" inherit_id="website.template_header_sales_four" active="False">
+    <xpath expr="//div[hasclass('o_main_nav')]" position="before">
+        <t t-set="_additional_btn_color" t-valuef="text-primary"/>
+    </xpath>
+</template>
+
+<template id="template_header_sidebar_additional_color_primary" inherit_id="website.template_header_sidebar" active="False">
+    <xpath expr="//div[hasclass('o_main_nav')]" position="before">
+        <t t-set="_additional_btn_color" t-valuef="btn btn-outline-primary"/>
+    </xpath>
+</template>
+
+<template id="template_header_boxed_additional_color_primary" inherit_id="website.template_header_boxed" active="False">
+    <xpath expr="//div[hasclass('o_main_nav')]" position="before">
+        <t t-set="_additional_btn_color" t-valuef="btn btn-outline-primary"/>
+    </xpath>
+</template>
```

Comment:
In fact, I think we could improve this and maybe solve the problem mentioned in my other comment at the same time: instead of making one template per header template so that the "primary" option does something, could not we do something like:

base template ->
```xml
<t t-if="add_additional_colors == 'primary'">
    <t t-set="_additional_btn_color" t-valuef="btn btn-outline-primary"/>
</t>
```
(using the most used occurence by default (btn-outline-primary))

Then your editor option would about enabling only one view: one that sets `add_additional_colors` to the right value to enable the option (with a xpath).

And for header templates that need to act differently than `btn-outline-primary` when the "primary" option is enabled you can xpath the `<t t-set="_additional_btn_color"` in the main template of that particular header.

Reply:
Ok I get it ! I've updated the branch 👍  : https://github.com/odoo/odoo/compare/43c2db5a1a34bc3f8fccaa076072da0210552fdb..dd8628e0762de50c127ed5fbbf64dd5354035330

What do you think ?